examples/tutorial/x/blog/handlers_test.go

```diff
@@ -1,14 +1,166 @@
+				Author: toWeaveAddress("3AFCDAB4CFBF066E959D139251C8F0EE91E99D5A"),
+			}),
+	}
+
+	objects = map[string]weave.Persistent{
+		"Blog": &Blog{
+			Title:       "this is a blog title",
+			NumArticles: 0,
+			Authors: [][]byte{
+				toWeaveAddress("3AFCDAB4CFBF066E959D139251C8F0EE91E99D5A"),
+			},
+		},
+		"Post": &Post{
+			Title:         "this is a post title",
+			Text:          "dLrvpy6wkdR2yG1ECKzxrQGeoVqTglnvQhk7Kagmiqu5c4AwszxIoB4FjGlUfjXyeNU5PCXJoqLFLYmiFnyW6ZvHYB3ZBczTydEJLd51f9bwtTuGhJ4P89vv7MsjTXMizERWm7KtQSiTBT9Vz6vTDmv5OrLoAHEfIK5wZXbhy8L9BNzjzV182Wvv7VdRd2iDid7cQW1FJX6PEgLYWG2A7wkUM76JaeeSlBvVdDtGLF5aav5eOYvVwzGC13SGQTnOKPBIhRFzX1o2g20kwpVxN0LDfm072UsY6Lx0DfDJnS5bvWsojim2BiPd8SjH0ChUN0NbuyUJhlDMfUPnM9LyDp31BXPSH4dRQcYpL4KPCugJ2t8oSqt4Arf2sAgMgdLipVdcm9qtbZuZRleCwT1ielP0jkk9pGFgyhJ7splO0UEDVJWBXvkAxs6fqrANMpQGoLUU7HPINuJbwXDG9208kXvWjNYBjLm8Yj0fosioTwXfNWxk7AvnwLkM1eXWhjBiKY91QA85THajebmwv5R4RS91RAeee67FjpkFT6d2rKDHWrKU5cxZbtPvKGR5Fk1V2mVlxPLoGAlMJmGXpjcDv78TJQQVPOYQbBqybmHlbLflulDRZeFT2VMARGsOsDWEjT3tDQ3NpJWlk4XiVhVJgR8Qy8oH3GGWLcjoCyMNHr5UMyLTYLwNpjxgmn8aoQHTg5m4gJtBWr0Rt82s5M6YQGpEjGfvhnIoEtzzZALjzlrVHAK13clM9ph28jrRXzCMzLANhvXQFoK0bLYoLNYTEPW0W6h1TNfq975hJlmshRhXBb19pyEk6YZ3LvapaZmSudE52t5iO91iXHl1ofDcQ7uTBypqhuRYpn41PA4QvzlE2M2ljIlKlw65n03JncOEVvqnucsDbn9XFjZBLOYrhytBwQuuSKggOudJBLWyz8UWn0XhE3uVt4jR7umX0HCXudeaLgXZvFpk7FjZsuXlBXr2Lffpj6yylBGN2gulHizZbRK7BFW7Py7dOw2VSuxG5bEVND6s4LgW9Imrdnku0TAJsdDahnbJ5t9IwvG0YmHOKLYWy49Se50FvoovsfodJUxfiyL8nYy86V8GgyhLKzDE2EfJWVmYroNErr5eJExBdcwg3WijMEXSxZXcRy3xFppuaTNxfmgiojk3ff8IR37YnAfHyfmPw6KuLupRH6ak12N6F2d7yrG0xno9eIoNpvGpMBpOWfNEajxFHM6i1C1bDHvVGfFwhpY5FdEsXIfXRetkiUNgbwnz36sQATlrj7B5FW6m5hL7yWp6mFI0xWtb0wdqaTIuJj08akHp6miWyWDDJHJrd5q2ipSfeJv0ZjSHqr51LBKZk3mW0r3aq28zBQatSgzQDwExeG2LeAPsQSnPiKUNzdJpONvoJv9ApwqOALD5cveakTzK9LQ5ZSl20uwx4N5JEYRdl2IZD1jgya54fk8wLGoNLlWHOqGrLdHru73nOGIHgGy8G4jhwNNsh2Vo",
+			Author:        toWeaveAddress("3AFCDAB4CFBF066E959D139251C8F0EE91E99D5A"),
+			CreationBlock: 100,
+		},
+	}
+
+	os.Exit(m.Run())
+}
+
 func TestCreateBlogMsgHandlerCheck(t *testing.T) {
+	db := store.MemStore()
+	tx := txs["CreateBlogMsg"]
+	handler := handlers["CreateBlogMsgHandler"]
+	res, err := handler.Check(weaveCtx, db, tx)
+
+	require.NoError(t, err)
+	require.Equal(
+		t,
+		newBlogCost,
+		res.GasAllocated,
+		fmt.Sprintf("gas allocated cost was equal to %d", res.GasAllocated))
 }
 
 func TestCreateBlogMsgHandlerDeliver(t *testing.T) {
+	db := store.MemStore()
+	tx := txs["CreateBlogMsg"]
+	handler := handlers["CreateBlogMsgHandler"]
+	_, err := handler.Deliver(weaveCtx, db, tx)
+	require.NoError(t, err)
+
+	expected := objects["Blog"]
+	bucket := handler.(CreateBlogMsgHandler).bucket
+	actual, err := bucket.Get(db, []byte("this_is_a_blog"))
```

Comment:
Yes, this is a good test.

I would add a few more cases here for this handler.Deliver:
* Create blog in empty store (done)
* Create blog with different name on non-empty store
* Create second blog with same name on non-empty store (should error)
* Create blog with missing title (should error)
* Create blog with context address != author (should error)

Reply:
For check, you can just get by with a positive case to make sure it isn't dropped... It is wrong to filter out valid data, just a little wasteful not to filter out invalid data (so better to err on side of being too permissive).